ccgateway/Dockerfile

```diff
@@ -1,12 +1,25 @@
-FROM ubuntu:16.04
+FROM ubuntu:18.04 as base
 
 WORKDIR /home/Creditcoin/Gateway
 
 COPY . /home/Creditcoin/Gateway/
 
+FROM base as dotnet
+
 RUN apt-get update
 RUN apt-get install -y wget apt-transport-https
-RUN wget -q https://packages.microsoft.com/config/ubuntu/16.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+RUN wget -q https://packages.microsoft.com/config/ubuntu/18.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
 RUN dpkg -i packages-microsoft-prod.deb
+RUN rm packages-microsoft-prod.deb
 RUN apt-get update
-RUN apt-get install -y dotnet-runtime-3.1
\ No newline at end of file
+RUN apt-get install -y dotnet-sdk-3.1
+
+FROM dotnet as build
+
+RUN dotnet build
+
+RUN ln -sr /home/Creditcoin/Gateway/bin/Debug/netcoreapp3.1/ccgateway .
+
+FROM build as gateway
```

Comment:
That would result in a larger than needed docker image. IMO the runtime container image should only have the binary artifacts and the dotnet-runtime deb package. The build container should still install the rest but later we discard it.

Reply:
FTR the dockerfile is run from the built output folder in the current CI pipeline:
`$(Build.SourcesDirectory)/ccgateway/bin/Release/netcoreapp$(netCoreVersion)/ubuntu-x64/Dockerfile`

It's just used to package the binaries and plugins currently